server/src/internalClusterTest/java/org/opensearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java

```diff
@@ -224,39 +222,73 @@ public void testIndexCreateBlockNotAppliedWhenAnyNodesBelowHighWatermark() throw
         assertFalse(state.blocks().hasGlobalBlockWithId(Metadata.CLUSTER_CREATE_INDEX_BLOCK.id()));
     }
 
-    public void testIndexCreateBlockIsRemovedWhenAnyNodesNotExceedHighWatermark() throws Exception {
+    public void testIndexCreateBlockIsRemovedWhenAnyNodesNotExceedHighWatermarkWithAutoReleaseEnabled() throws Exception {
         final Settings settings = Settings.builder()
             .put(DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_DISK_THRESHOLD_ENABLED_SETTING.getKey(), false)
             .build();
 
         internalCluster().startClusterManagerOnlyNode(settings);
-        final List<String> dataNodeNames = internalCluster().startDataOnlyNodes(2, settings);
-        final List<String> indexNames = new ArrayList<>();
+        internalCluster().startDataOnlyNodes(2, settings);
         ensureStableCluster(3);
 
+        final MockInternalClusterInfoService clusterInfoService = getMockInternalClusterInfoService();
         // Reduce disk space of all node until all of them is breaching high disk watermark.
-        for (final String dataNodeName : dataNodeNames) {
-            final String indexName = populateNode(dataNodeName);
-            indexNames.add(indexName);
-        }
+        clusterInfoService.setDiskUsageFunctionAndRefresh(
+            (discoveryNode, fsInfoPath) -> setDiskUsage(fsInfoPath, TOTAL_SPACE_BYTES, WATERMARK_BYTES - 1)
+        );
 
-        getMockInternalClusterInfoService().refresh();
         // Validate if cluster block is applied on the cluster
         assertBusy(() -> {
             ClusterState state = client().admin().cluster().prepareState().setLocal(true).get().getState();
             assertTrue(state.blocks().hasGlobalBlockWithId(Metadata.CLUSTER_CREATE_INDEX_BLOCK.id()));
         }, 30L, TimeUnit.SECONDS);
 
-        // Delete indices to free space
-        deleteIndices(indexNames);
-        getMockInternalClusterInfoService().refresh();
+        // Free all the space
+        clusterInfoService.setDiskUsageFunctionAndRefresh(
+            (discoveryNode, fsInfoPath) -> setDiskUsage(fsInfoPath, TOTAL_SPACE_BYTES, TOTAL_SPACE_BYTES)
+        );
+
         // Validate if index create block is removed on the cluster
         assertBusy(() -> {
             ClusterState state1 = client().admin().cluster().prepareState().setLocal(true).get().getState();
             assertFalse(state1.blocks().hasGlobalBlockWithId(Metadata.CLUSTER_CREATE_INDEX_BLOCK.id()));
         }, 30L, TimeUnit.SECONDS);
     }
 
+    public void testIndexCreateBlockIsRemovedWhenAnyNodesNotExceedHighWatermarkWithAutoReleaseDisabled() throws Exception {
```

Comment:
You can try to add this scenario.
1. Index create block added by User with autoReleaseDisabled
2. WaterMarkBreaches and Decider also puts Index create block.
3. User release the index block
4. Check that index create block re applies on the cluster by Decider